mozilla/fxa
# feat(settings): Account recovery key PDF download

## Because

* We want to include more information in the recovery key file, and rendering for localized content in .txt format was inconsistent across apps and platforms.

## This pull request

* Add react-pdf library and associated dependencies for client-side PDF generation
* Add DownloadRecoveryKeyPDF component to handle PDF generation and download
* Create RecoveryKeyPDF with styling and l10n
* Update tests for recovery key
* Add pdf-parse library to test PDF content with Playwright
* Add font assets to [fxa-settings](https://mozilla-hub.atlassian.net/browse/FXA-settings) for PDF creation
* Remove fonts from CDN assets folder (no longer needed)

## Issue that this pull request solves

Closes #[FXA-7223](https://mozilla-hub.atlassian.net/browse/FXA-7223)

## Checklist

_Put an `x` in the boxes that apply_


## Screenshots (Optional)

![image](https://github.com/mozilla/fxa/assets/22231637/e0d5a95d-10df-4378-b2cb-ece9e6f8d6f3)

PDFs are only partially localized - most strings are new

![image](https://github.com/mozilla/fxa/assets/22231637/ebd3317c-2c84-4a47-8a3e-cf7d67a8816f)

With RTL layout
![image](https://github.com/mozilla/fxa/assets/22231637/c0092d74-fd66-4465-ba24-11e2ee0e2275)

## Other information (Optional)

PDF download can be tested from:

1. the `ButtonDownloadRecoveryKeyPDF` in Storybook (change preferred language from browser settings to test localized downloads)
OR

2. the recovery key creation flow on localhost (will need to append `&isInRecoveryKeyExperiment=true` param to see the new recovery key flow in `/settings`

[FXA-7223]: https://mozilla-hub.atlassian.net/browse/FXA-7223?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
